lib/api/completeMultipartUpload.js

```diff
@@ -361,15 +383,19 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                             .getSerializedUids()));
                     }
                     return next(null, mpuBucket, mpuOverviewKey,
-                        aggregateETag, storedPartsAsObjects);
+                        aggregateETag, storedPartsAsObjects,
+                        extraPartLocations);
                 });
         },
         function waterfall8(mpuBucket, mpuOverviewKey, aggregateETag,
-            storedPartsAsObjects, next) {
+            storedPartsAsObjects, extraPartLocations, next) {
             const keysToDelete = storedPartsAsObjects.map(item => item.key);
             keysToDelete.push(mpuOverviewKey);
             services.batchDeleteObjectMetadata(mpuBucket.getName(),
                 keysToDelete, log, err => next(err, aggregateETag));
+            data.batchDelete(extraPartLocations, logger
```

Comment:
minor: we could check if extraPartLocations is an empty array to avoid making an unnecessary function call & extra log entry in batchDelete with no keys to delete